libs/prelude/Language/Reflection.idr

```diff
@@ -396,9 +396,9 @@ mutual
   withNInj Refl = (Refl, Refl)
 
   private
-  instanceNInj : (InstanceN n xs = InstanceN n' ys) -> (n = n', xs = ys)
-  instanceNInj Refl = (Refl, Refl)
-
+  implementationNInj : (ImplementationN n xs = ImplementationN n' ys) -> (n = n', xs = ys)
```

Comment:
@ahmadsalim one thing is confusing me that if a deprecations occurs in Idris code i can use a `%deprecated` annotation and if a deprecation come from parser i can also use `<|>` to tackle with it, but how can i mark the deprecated constructors in Haskell data types like `InstanceN` you mentioned, i mean, these constructors are internal and will not be exposed to users, there is also not exist something like `deprecated` anntation in Haskell, so maybe simply add note to `CHANGELOG` is enough? i'll address deprecations in Idris lib files and Parser first.